packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -12,34 +12,49 @@ import {
   GraphObjectStore,
 } from '@jupiterone/integration-sdk-core';
 
-import { BucketMap } from './BucketMap';
 import { flushDataToDisk } from './flushDataToDisk';
 import {
   iterateEntityTypeIndex,
   iterateRelationshipTypeIndex,
 } from './indices';
+import { InMemoryGraphObjectStore } from '../memory';
 
-export const GRAPH_OBJECT_BUFFER_THRESHOLD = 500; // arbitrarily selected, subject to tuning
+export const DEFAULT_GRAPH_OBJECT_BUFFER_THRESHOLD = 500;
 
 // it is important that this value is set to 1
 // to ensure that only one operation can be performed at a time.
 const BINARY_SEMAPHORE_CONCURRENCY = 1;
 
+export interface FileSystemGraphObjectStoreParams {
+  /**
+   * The maximum number of graph objects that this store can buffer in memory
+   * before writing to disk. Machines with more memory should consider bumping
+   * this value up.
+   *
+   * Default: 500
+   */
+  graphObjectBufferThreshold: number;
+}
+
 export class FileSystemGraphObjectStore implements GraphObjectStore {
-  semaphore: Sema;
-  entityStorageMap: BucketMap<Entity>;
-  relationshipStorageMap: BucketMap<Relationship>;
+  private readonly semaphore: Sema;
+  private readonly localGraphObjectStore = new InMemoryGraphObjectStore();
+  private readonly graphObjectBufferThreshold: number;
 
-  constructor() {
-    this.entityStorageMap = new BucketMap();
-    this.relationshipStorageMap = new BucketMap();
+  constructor(params?: FileSystemGraphObjectStoreParams) {
     this.semaphore = new Sema(BINARY_SEMAPHORE_CONCURRENCY);
+    this.graphObjectBufferThreshold =
+      params?.graphObjectBufferThreshold ||
+      DEFAULT_GRAPH_OBJECT_BUFFER_THRESHOLD;
   }
 
   async addEntities(storageDirectoryPath: string, newEntities: Entity[]) {
-    this.entityStorageMap.add(storageDirectoryPath, newEntities);
+    this.localGraphObjectStore.addEntities(storageDirectoryPath, newEntities);
 
-    if (this.entityStorageMap.totalItemCount >= GRAPH_OBJECT_BUFFER_THRESHOLD) {
+    if (
+      this.localGraphObjectStore.getTotalEntityItemCount() >=
+      this.graphObjectBufferThreshold
```

Comment:
IMO `graphObjectBufferThreshold` makes me think that the `localGraphObjectStore` can store 500 graph objects, but really it can store 500 of each class of graph object (500 entities & 500 relationships). Just curious if it's worth maintaining a `private readonly entityBufferThreshold: number;` and separately a `private readonly relationshipBufferThreshold: number;`, or else update docstrings on `FileSystemGrahpObjectStoreParams.graphObjectBufferThreshold` to reflect that totals for entities and relationships are kept separate?

Reply:
Perhaps we can consider introducing that level of granularity later if necessary. I don't think it's worth maintaining the two properties for now though. I imagine that relationships generally consume less memory than entities, so it may be worth considering later.